### ISSUE TITLE
Replace Awesomium

### ISSUE THREAD

**Issue by DarthTealc** [Facepunch/garrysmod-requests#531]
Awesomium is based on Chrome/Chromium 18 AFAIK. That's way too old. And there's no indication they're going to fix that any time soon.

Can Awesomium please be replaced with a more up-to-date browser implementation? For example "Chromium Embedded Framework" which is used in Steam.

**Comment by xaviergmail** [comment]
I would love for better HTML panel support (As well as the loading screen, because I can't use Canvas in it right now and it makes me very sad)

**Comment by P4sca1** [comment]
It seems like everybody is complaining about Awesonium since months... 
I think there should finally be done something.

**Comment by Kefta** [comment]
It's not that easy, though. It's going to be a long process to use the new API if it's done at all

**Comment by HandsomeMatt** [comment]
coherent ui is the way to go, easy to swap out for awesomium and Facepunch already has the licensing from Rust I'm psure

**Comment by P4sca1** [comment]
This sounds as it will be changed in the near feature. Good job! :D

**Comment by wiox** [comment]
> This sounds as it will be changed in the near feature. Good job! :D

:rainbow:

**Comment by Noiwex** [comment]
I want to apply for job position in facepunch only because of issues like
this. Seems like no one else doesn't even bother to do a thing.

**Comment by WinterPhoenix96** [comment]
Cinema is no longer able to play YouTube Videos at above 360p quality due to the limitations of Awesomium. [More info here.](http://facepunch.com/showthread.php?t=1450721&p=48352028&viewfull=1#post48352028)

Please, please, sooner rather than later, do this.

**Comment by Noiwex** [comment]
Garry pls hire me :v:

**Comment by crazyscouter** [comment]
best application 10/10

**Comment by aStonedPenguin** [comment]
+1

**Comment by Authorjames** [comment]
:+1:

**Comment by SwadicalRag** [comment]
:100:+

**Comment by poseidonservers** [comment]
Definitely past due, Cinema has become a big part of Garrysmod. ༼ つ ◕_◕ ༽つ Give Replacement

**Comment by Splerge** [comment]
+1, don't like Awesomium, a new framework would be better.

**Comment by kklouzal** [comment]
+1 Replace Awesomium. Use Steam's framework.

**Comment by Jake3231** [comment]
I agree. Along with insufficient support for different things, Awesomium is very unstable and is often the cause of crashes in my game.

**Comment by gamerpaddy** [comment]
+1 Replace Awesomium, its crashing my game too frequently.

**Comment by WinterPhoenix96** [comment]
Posting for relevance. https://facepunch.com/showthread.php?t=1485127&p=48950004&viewfull=1#post48950004

**Comment by HDG-MrDeeJay** [comment]
did I +1 this thing yet? Awesomium is the cancer that's killing GMOD,

Followed closely by Scriptfodder and Skidcheck.

**Comment by AnonTakesOver** [comment]
+1

**Comment by Sophie-bear** [comment]
+1 from me as well. There's really no good reason to use such an outdated and dying API.

**Comment by Chewgum** [comment]
replace the shit already

**Comment by HDG-MrDeeJay** [comment]
*COUGH*

**Comment by Kefta** [comment]
Stop spamming; it's already being discussed in the Next Update thread

**Comment by aStonedPenguin** [comment]
no

**Comment by jasmin12e** [comment]
Please :( I don't want to see cinema just die off..

**Comment by xaviergmail** [comment]
This has nothing to do with *JUST* cinema. There are many great uses for Awesomium, including but not limited to HTML UI's, which would make many people's lives easier rather than dealing with horrible derma. And please don't say docking. Making responsive UI's that scale well with docking is a _pain_. It would also help improve our loading screens with cool canvas stuff. I have a beautiful loading screen just sitting in the shitter right now, waiting to see the light of day, and I'm sure many other people have many other cool things waiting. Not to mention the huge performance impact that even further limits what we can do with Lua. Not saying anything else would be performance-tastic, but I would hope so. Rant over.

**Comment by HDG-MrDeeJay** [comment]
Yeah, I too have a loading screen that has had to be dumbed down because 
Awesomium just couldn't handle it. It's slow, it's old, and any delays 
in its replacement will only do further to harm the community, the very 
community that has kept this game strong for how many years?

You can claim "its too hard" or "i dont have the testing environments" 
all you like. The fact is, Awesomium MUST be replaced, and the sooner it 
gets done, the better everyone's lives will be.

**Comment by jackson-c** [comment]
JUST DO IT

**Comment by ILOVEPIE** [comment]
Awsomium needs to be dropped, my suggestion would be to go with the open source alternative to awsomium: https://bitbucket.org/chromiumembedded/cef

**Comment by zerfgog** [comment]
awesomium wraps cef by the way

**Comment by DuckyC** [comment]
+1

**Comment by ILOVEPIE** [comment]
Awesomeium does not wrap CEF at all

**Comment by ILOVEPIE** [comment]
https://html5test.com/s/f8cfde2afcf3ba18.html <- there's your problem (this is the score page for the version of Awsomium currently in the dev branch of gmod)

**Comment by HDG-MrDeeJay** [comment]
>no mp3 audio support

Come on facepunch, step it up!

**Comment by WinterPhoenix96** [comment]
I've made another post arguing for Awesomium's replacement, highlighting most of the reasons for why it should be done, [here](https://facepunch.com/showthread.php?t=1497739&p=49553475&viewfull=1#post49553475).

**Comment by HDG-MrDeeJay** [comment]
Dont waste your time. They will just update awesomium to some pointless 
build that fixes absolutely nothing.

**Comment by meepdarknessmeep** [comment]
:-1: :box :-1:

**Comment by Velkon** [comment]
You're a pointless build that fixes absolutely nothing!

**Comment by HDG-MrDeeJay** [comment]
So is your mother.

**Comment by Velkon** [comment]
My mother atleast fixed her mistake, unlike yours!!!!

**Comment by HDG-MrDeeJay** [comment]
she gave you up for adoption?9

**Comment by zerfgog** [comment]
skid fight! skid fight!

**Comment by Velkon** [comment]
Shut up skid

**Comment by HDG-MrDeeJay** [comment]
ok, i can tell this is going beyond a harmless squabble so im stopping 
this before someone else comes in and gets mad.

**Comment by kalis0x** [comment]
ALL RIGHT WHAT THE FUCK DO YOU THINK YOU'RE TALKING ABOUT YOU FUCKING BITCH
THAT'S NOT HOW IT FUCKING WORKS YOU CAN'T JUST REPLACE IT LIKE THAT
FUCKING SKID KILL YOURSELF

**Comment by HDG-MrDeeJay** [comment]
shit too late xD

**Comment by kalis0x** [comment]
YOU THINK THIS IS FUCKING FUNNY, SO YOU PUT YOUR SHITTY "XD'S"
WHAT A FUCKING INCOMPETENT 12 YEAR OLD LOSER

**Comment by HDG-MrDeeJay** [comment]
kalis0x are you from mcdonalds you are putting too much salt on those fries.

**Comment by kalis0x** [comment]
unfunny

:-1:

**Comment by HDG-MrDeeJay** [comment]
Nah its funny if you were at maccas with me the other day. Their chips were so salty i couldn't taste them through the salt.

**Comment by kalis0x** [comment]
4:39 PM - $: Replace Awesomium
https://github.com/Facepunch/garrysmod-requests/issues/531
4:39 PM - Velkon.: "ok, i can tell this is going beyond a harmless squabble so im stopping
this before someone else comes in and gets mad."
4:40 PM - Velkon.: Someone storm in
4:40 PM - Velkon.: all caps

**Comment by HDG-MrDeeJay** [comment]
I kinda figured.

**Comment by Kefta** [comment]
Oh my god shut the fuck up you all are acting like cdzira or whatever his name is

**Comment by HDG-MrDeeJay** [comment]
sorry kefta

**Comment by aStonedPenguin** [comment]
yeah fuck that cdriza skid dont be like him

**Comment by SwadicalRag** [comment]
:point_down: :point_up_2: 
:point_left: :point_right:

**Comment by HDG-MrDeeJay** [comment]
:point_up_2::point_up_2::point_down::point_down::point_left::point_right::point_left::point_right: B A Start

**Comment by OGCristofer** [comment]
Please fix this, Cinema servers are becoming slower and are lagging more thanks to this.

**Comment by WinterPhoenix96** [comment]
It seems as though @willox has stepped up to the plate! https://facepunch.com/showthread.php?t=1497739&p=49580543&viewfull=1#post49580543

**Comment by Velkon** [comment]
:late:

**Comment by rafitamolin** [comment]
PLEASE

**Comment by WinterPhoenix96** [comment]
Cinema can no longer use the old (deprecated) YouTube player reliably to get around videos having a maximum resolution of 360p. Which means we're now effectively stuck at 360p/stuck with that weird 3D glitch that sometimes happens as a result for YouTube, unless Awesomium is replaced.

More here: https://facepunch.com/showthread.php?t=1516664&p=50564400#post50564400

**Comment by Shigbeard** [comment]
Yes, I think its time to bring this issue back to the spotlight. Bumping this for more attention.

**Comment by jackson-c** [comment]
Needs to be fixed ASAP. YouTube is on the edge of completely ceasing to work, which breaks numerous addons as well as the entire *very popular* cinema gamemode.

**Comment by meepdarknessmeep** [comment]
Wait, if youtube breaks this means an end to bad youtube music on loading screens! WE MUST STAY HOW IT IS!

**Comment by Shigbeard** [comment]
It wont fix it, you can still play audio files anyway. Better to force 
mute audio in loading screens.

**Comment by Digiquo** [comment]
If Cerealbox is kill, we can't get our dank memes. #replaceawesomium

**Comment by garrynewman** [comment]
This ain't gonna happen

**Comment by rafitamolin** [comment]
savage asf

**Comment by zerfgog** [comment]
@garrynewman Do you have any specific reasons for this? As far as I've seen @willox has some [really](https://facepunch.com/showthread.php?t=1497739&p=49602588&viewfull=1#post49602588) great progress with his CEF implementation.

I mean look at [this](http://scrnsht.me/u/thb/raw) @garrynewman !

It's really unfortunate that we have this really big limitation in an otherwise extraordinarily flexible platform.

**Comment by aStonedPenguin** [comment]
> This ain't gonna happen

im glad you care

**Comment by comentarinformal** [comment]
So much for the guy who, four days ago, said on his blog 

> When we release a game **_our long term plan is to update it and improve it_**. This works because our games are all multiplayer or sandbox games. And so far it’s turned out to be a financially sound strategy, with our games eventually selling more and more each year.

**Comment by Revenge282** [comment]
Fucking top notch development oversight.  Was doing much better before.

**Comment by SnowboiTheGr8** [comment]
@garrynewman who are you?

**Comment by jackson-c** [comment]
![maxresdefault 1](https://cloud.githubusercontent.com/assets/8286907/16286106/1d9f82fc-388f-11e6-82dd-edc78211600e.jpg)

**Comment by Shigbeard** [comment]
Holy shit did garry literally just say "fuck you we aint fixing something thats broken"?
I think I now understand why Rust is still in early access.

**Comment by Kefta** [comment]
Because insulting him will change everything

**Comment by Shigbeard** [comment]
I've actually been editing my message for the past 10 minutes @Kefta Kefta

**Comment by SnowboiTheGr8** [comment]
@Shigbeard preach it

**Comment by Sophie-bear** [comment]
I wish we'd have gotten a reason at the very least to explain this decision. If, and it's a big if, Awesomium ever gets updated to a reasonably recent version of Chromium, that'd be a fine alternative too, but the reason we all are for CEF is that it is maintained and kept up-to-date; something the developers behind Awesomium haven't been known for lately...

**Comment by Noiwex** [comment]
well, fuck, time to remove playx from server i guess

**Comment by jackson-c** [comment]
Alternate idea: bug, or possibly pay, the developers of awesomium to update their platform.

**Comment by Kefta** [comment]
Or just contribute to the CEF module

**Comment by Shigbeard** [comment]
The one that keeps crashing?

**Comment by Kefta** [comment]
Yes

**Comment by Bo98** [comment]
> Alternate idea: bug, or possibly pay, the developers of awesomium to update their platform.

They claimed that they would release v1.7.6 in "a few days". That was 3 weeks ago though.

Though I won't be surprised if its Chromium backend is not getting updated until v2.0 - a version which they've been claiming to be working on for over a year. But remember, if it updates too far then we won't even be able to use it due to the lack of OS X 32-bit support and XP + Vista support as I doubt Awesomium will add that back into the Chromium backend.

**Comment by newfulluser** [comment]
Vimeo stopped working, and Blip is now redirecting to maker.tv which is not supported by cinema. 
Is there a way to play video files stored on my shared hosting account? Because I've tried several times and the button never gets red on cinema. Thanks

**Comment by neico** [comment]
This isn't really the right place for cinema gamemode related support...
https://github.com/pixeltailgames/cinema

**Comment by newfulluser** [comment]
@neico Well it's mostly dead there, and if the browser was changed to chromium vimeo would work again.

**Comment by neico** [comment]
that's not really what you were talking about though. And as long as @willox doesn't come up with something in this regard (be it a possible alternative like cef of steam's html surface, or even just a message that overthrows garry's statement) it won't happen...

**Comment by Shigbeard** [comment]
Unless willox magically makes CEF cheap and 32 bit OS X compatible, it 
won't happen.

**Comment by gamerpaddy** [comment]
Stfu fuckers

**Comment by uRandomAlex** [comment]
Just tell all your players to install this: https://github.com/lunation/Awesomium-Killer-2.0 .
Currently it crashes your game every hour or two, but you can always contribute.

**Comment by uRandomAlex** [comment]
Also, implementing Steam's HTTP as an alternative to Awesomium is a good idea.
Help us, our god @willox !

**Comment by neico** [comment]
I actually tried to tackle the steam api one a while ago, even with some help I wasn't able to get it to render onto the vgui texture, might be partly because some sdk interfaces were changed in gmod (that would require some serious reverse engineering to fix), might be because I fucked something up somewhere.

Also, imo such "killers" are cancer, first of it's replacing a dll with fake content (in this case replacing one infrastructure with a different one, most likely causing all sorts of clashes), making the already unstable html stuff even more wibble-wobble. Then there's the aspect of it being killed of by verify steam cache and possibly updates. And maintainable code looks different...

Then there's also the aspect that raw CEF can't play proprietary codecs like h264 or mp3 due to licensing yatta yatta. Steam's HTMLSurface has those licenses covered though and therefore has native access to those codecs so it's the best way to go regarding things like cinema.

**Comment by uRandomAlex** [comment]
@neico Of course Steam HTML will be better, but currently we're stuck with CEF by this cool guy. YouTube already provides .webm videos, so we can safely play them using CEF.

**Comment by SpencerSharkey** [comment]
this is pathetic

**Comment by Noiwex** [comment]
this is garry

**Comment by Shigbeard** [comment]
I've said it before and I'll keep on saying it. Garry's only mission in life regarding garry's mod is to come back from rust every 3 to 6 months to take a massive dump on whoever is actually doing a massive project on gmod to change it (such as server browser or awsomium) to the tune of "this aint happening"

**Comment by Adam759** [comment]
Now sitting here in Feb 2017 still waiting for at least HTML5 support. With the way things are going HTML5 will be everywhere, and already is. Is it really that much to ask for to update it to support something so crucial and current?

**Comment by Rusketh** [comment]
This would only benefit the community over all. Can somebody from face punch just get this done and lets move on with our lives. PLEASE!

**Comment by Shigbeard** [comment]
It won't happen because Badintosh users, or to be more precise: "Mac OS 
doesn't support CEF, and as gmod is advertised as available on Mac, a 
core feature that isn't supported on Mac cannot be added to the game."

I believe it has something to do with 64bit.

**Comment by Jake3231** [comment]
I personally play Garry's Mod on my Mac, and supporting only 64-bit is an issue.  To be fair, it's not the fault of the users, but it is annoying that the Mac version of the game does not support 64-bit.  This has caused the game to hit it's memory limit quite a few times, and now it seems that it can't even support CEF.  I agree that we need something new, but this brings up a whole new issue about improving the Mac's compatibility.

**Comment by Adam759** [comment]
There has to be something that is cross compatible with Mac and Windows that can be integrated. For how many issues there are right now I am astonished this issue hasn't been looked at more.

**Comment by Turbine1991** [comment]
Could we please get some fresh eyes on this issue?

Chrome 18 is dated (2012), this thread is from 2015 - The web has crept on a lot since then.

When one has to employ workarounds (albeit simple) to get some fundamental web advances such as 'display: flex;' to work, you know this is a pressing issue. 49 stable milestone releases have been pushed since.

I cannot stress enough, please spend the time to replace Awesomium, it's one change which can be justified.

**Comment by handsomematt** [comment]
https://gmod.facepunch.com/commits/

There is a work in progress branch to replace Awesomium.

**Comment by Ajbolitt76** [comment]
@willox will facepunch replace awesomium at next update? The chromium branch looks stable.

**Comment by Noiwex** [comment]
![](https://i.imgur.com/y8Tgzgc.png)

Don't think it's good

**Comment by Sophie-bear** [comment]
Chromium is updated and maintained in the same branch as 64-bit builds now from what I can tell from commit logs. https://gmod.facepunch.com/commits
This is not something we have access to yet to really know for sure what it's about, so I guess we'll see.

**Comment by Fabiassassin** [comment]
> 
> 
> ![keemstar](https://cloud.githubusercontent.com/assets/8286907/16286316/8fcc1704-3890-11e6-83e8-5b5617504519.jpg)
> 
> Garry will change his mind when YouTube ceases to work entirely.

This person predicted the future; YouTube does not work anymore

**Comment by Noiwex** [comment]
![](https://cdn.discordapp.com/attachments/265960516634148864/491956875831803904/unknown.png)

RIP

**Comment by LewisTehMinerz** [comment]
press ❤️ to pay respects to media player

**Comment by Rusketh** [comment]
Just done a load of work with html and java script for some GUI's, turns out GarrysMod is not update enough to display them correctly. Can we please get this updated.

**Comment by Noiwex** [comment]
You've been making GUI for GMod without testing? 🤔

**Comment by LewisTehMinerz** [comment]
:eyes:

**Comment by chelog** [comment]
Almost 4 years from this issue created and still no sign of Chromium on main branch. This cannot be serious, guys. It's pretty stable now to replace Awesomium. Even if Chromium now has some bugs or incompletes in gmod, it'd be MUCH more useful than having this old broken thing in the game. Start pushing it to main, please

**Comment by Noiwex** [comment]
Lol, no, it's not stable, it crashes players a lot.

**Comment by VaasKahnGrim** [comment]
is there by chance a trello page or something similar that facepunch uses for these types of things? Would help in knowing how far along features like this are

**Comment by Bo98** [comment]
Unfortunately not. The status of it now is that development is taking place on the 64-bit branch. This move was essential in order for Chromium to be supported on all platforms.

How far away is it? I honestly don't know, but it's one of the main focusses.

For the 32-bit Chromium branch, I wouldn't expect much further development there.

**Comment by WinterPhoenix96** [comment]
macOS still isn't supported on the 64-bit branch and that needs to be done before they can push it to main.

**Comment by Bo98** [comment]
Yep, that's what's being worked on just now.

**Comment by chelog** [comment]
@Noiwex actually it crashes only when being used :/

People who are on chromium branch do not crash at all in my servers (the ones that don't utilize chromium), however yes, when used, it does crash (which by the way didn't happen at all until recent update like around 2 weeks ago). Oh cmon, half of gmod servers already crashing clients by things that are in master branch for a long time

**Comment by WinterPhoenix96** [comment]
Hot take:
Creating issues for crash dumps including any information about what was going on when the crash took place helps the Facepunch developers identify and fix them, so they don't happen anymore.

They don't magically go away if they don't know what to look for and fix. Players have to give them that information.

**Comment by Sophie-bear** [comment]
People on Windows should probably be using the x86-x64 branch for Chromium support rather than the Chromium branch anyway, at this point. See if they experience the same crashes on there.

**Comment by Noiwex** [comment]
The thing is: it crashes without dumps

**Comment by owltaver** [comment]
With such a speed of developmen, it seems to me by the time when the framework is beeing updated, it will be outdated like awesomium.) Already, it looks old against the new versions of chromium, most types of playback are not supported. We can forget about mp4 and new streaming format witсh used on youtube and some new services. So yeah, i just hope chromium will be upgraded to a newer version at least after it comes out of the beta version. Otherwise, sense of this update will be quite a bit.

![3oGQV58](https://user-images.githubusercontent.com/52373664/60395679-c3b25500-9b3f-11e9-85b5-4b3b66798983.png)

**Comment by WinterPhoenix96** [comment]
Chromium will not support Proprietary Codecs in GMod no matter the version. Licensing is an issue for Facepunch.

However, Solstice Game Studios and I have made a workaround for that.

Introducing GModCEFCodecFix: https://github.com/solsticegamestudios/GModCEFCodecFix/releases

**Comment by handsomematt** [comment]
wow thank u solstaice gam studios !!

**Comment by owltaver** [comment]
> Chromium will not support Proprietary Codecs in GMod no matter the version. Licensing them is an issue for Facepunch.
> 
> However, Solstice Game Studios and I have made a workaround for that.
> 
> **Introducing GModCEFCodecFix:**
> https://github.com/solsticegamestudios/GModCEFCodecFix/

Wow, now i know. Thank you for that!

**Comment by benrob0329** [comment]
Doesn't YouTube support VP8/9 though (at least for some videos)?

**Comment by WinterPhoenix96** [comment]
> Doesn't YouTube support VP8/9 though (at least for some videos)?

It supports VP9 for **most** videos except those that are/were Live.

VP8 was dropped around September 2018 which made everything on YouTube stop working on Awesomium. Scroll up for more info on that.

Nearly every other video service uses H.264 (MP4) in some way, however, even Twitch. Which is why proprietary codecs are still an issue.

**Comment by AnthonyDMolnar** [comment]
I tried to download the latest branch at defender flagged it as a virus and will not allow it to be downloaded.  Anyone have an explanation on that?

**Comment by VaasKahnGrim** [comment]
> I tried to download the latest branch at defender flagged it as a virus and will not allow it to be downloaded. Anyone have an explanation on that?

Disable Windows Defender Bloatware

**Comment by AnthonyDMolnar** [comment]
While I support the idea, it seems like a pretty big risk to do that.  What is in there that is flagging it as a virus?

**Comment by VaasKahnGrim** [comment]
I had a similar issue with an old antivirus called Avast! it had been flagging html stuff for garrysmod as malicious. it could be something similar

**Comment by AnthonyDMolnar** [comment]
Okay, that makes some sense.  I'm currently waiting out my chat delay timer on Solstice's discord so I can inquire more about the details from one of the developers.  I do other coding work and I need to be cautious about any virus flags.  I find it odd because I've downloaded tons of HTML based work from Git before and never had this issue.

**Comment by Sophie-bear** [comment]
The official answer from WinterPhoenix about our "GMod CEF Codec Fix" tripping AV's:
"It trips heuristics because it does things that are "suspicious," like downloading files from the internet and then overwriting existing files.
The source code is available if you want to review it, but you'll see it doesn't do anything other than what's it's described to do."

**Comment by DarthTealc** [comment]
Is CEF still happening?

Awesomium is based on Chromium 17 as far as I was able to determine. Chrome 17 released 2012-02-08 which means Gmod is now using a browser that hasn't been updated in 10 years and is missing 10 years worth of standards changes, performance improvements, security fixes, etc. 
Happy Birthday Awesomium 🎉

**Comment by Kefta** [comment]
>he doesn't know

**Comment by VaasKahnGrim** [comment]
Could probably find the source code to awesomium and just use an updated version of it. Valve's site has a whole thing about it apparently https://developer.valvesoftware.com/wiki/Implementing_Awesomium

Either that or prey rubat sees this and does it for us. or just wait for Chromium portion to get put in with the x86_64 branch when that becomes the main branch

**Comment by ILOVEPIE** [comment]
This has already been taken care of they already switched to CEF

**Comment by WinterPhoenix** [comment]
Use the x86-64 branch

**Comment by VaasKahnGrim** [comment]
Ah good, about time that was fixed
and not until x86-64 is the main branch :p

**Comment by AusHick** [comment]
> Use the x86-64 branch

How long have we had a x64 branch now though? Same thing with multi core rendering forever ago.

There's no reason to expect anything new to Garry's Mod besides the bare minimum of life support, in my opinion. It's a shame but it's EOL with Sandbox on the horizon.

**Comment by DarthTealc** [comment]
Happy 11th Birthday Awesomium!

... So when is the x64 branch being moved to main? Does _anyone_ work on Gmod any more? Will it take a hostile decade-old web exploit to get Awesomium replaced? 

Gmod just needs Awesomium replaced and a fixed server list in order to survive. It's completely dead in various countries due to fake servers and a terrible handling of gamemode listing.

**Comment by DarthTealc** [comment]
> macOS still isn't supported on the 64-bit branch and that needs to be done before they can push it to main.

Drop Mac support or give it a barebones fallback menu not using a web renderer so it can still run on 32bit Mac. 

Most Valve games don't even support current versions of MacOS due to Apple's actively hostile behaviour towards decades of 32-bit games. Mac OS account for 2.61% of Steam users, of which at least 70% can't run Gmod already as they're on 64-bit only versions. Why are we maintaining support for a maximum of 0.78% of Steam users, a number that will only keep dropping, at the expense of the nearly 100% of Windows and Linux users who have to keep using Awesomium?

**Comment by AkikoKumagara** [comment]
> >
> Drop Mac support or give it a barebones fallback menu not using a web renderer so it can still run on 32bit Mac.

In my own experience, the x86-64 branch generally works on macOS currently, even on the latest versions. I've heard people having issues with the menu not loading (which is apparently something some sometimes happens on a vanilla install), but paired with SGS's [https://github.com/solsticegamestudios/GModCEFCodecFix](GModCEFCodecFix), I've been unable to reproduce them with said fix and no one has provided logs to us after being asked, still.
While it sucks that we need third-party solutions for the game still, it's not in a completely broken state like your comment may have people believe; it just doesn't work without _at least_ hopping into a beta branch for most affected players.
We have already added one fix for the menu failing to load in w/ GMod CEF Codec Fix.

**Comment by DarthTealc** [comment]
> In my own experience, the x86-64 branch generally works on macOS currently, even on the latest versions.

In that case Gmod could push x86-64 to the main branch on all platforms. If there are remaining issues with the x86-64 branch for Mac users, making it default will bring them out of the woodwork and then the bugs can be fixed. A beta branch for 32bit with Awesomium could be made available for a while for users who experience issues until those are fixed.

**Comment by DarthTealc** [comment]
Could we please get confirmation that CEF/x86-64's release isn't going to require it to be absolutely perfect in every single way and solve world hunger? Cause that's never going to happen, in any game or any update. It just needs to be better than what Gmod currently has, and it seems like GModCEFCodecFix's fixes bring this branch to an overall better state than current. The only notable outstanding issue I can see is the Linux multi-monitor resolution bug ([issue #5326](https://github.com/Facepunch/garrysmod-issues/issues/5326)), nobody has mentioned any other unfixed issues.

Please [don't let perfection get in the way of improvement](https://en.wikipedia.org/wiki/Perfect_is_the_enemy_of_good) or we'll be suck on bad forever.

**Comment by DarthTealc** [comment]
What is the appropriate location to have discussions about CEF and general questions about what is holding Facepunch staff back from releasing it, so we can help test and report issues and make some progress? A discord moderator shuts me down when I participate in a discussion that someone else brings up (though doesn't tell off the person who started the discussion, just me), and repeatedly tells me to "give up" without providing any evidence that Facepunch has completely abandoned Gmod (what they appear to be implying Gmod's state is and claim I "know the answer already", which I don't, as nobody at Facepunch has announced Gmod's cancellation), and calling me a troll. I feel this is unwarranted and I do not understand why I am being targeted to the exclusion of others. I don't even know if I'm allowed to participate in any discussion or if it's just CEF I'm targeted for. I may not be prolific but I have a long history with Gmod since 2007 and various iterations of the community and have never caused trouble.

I have no prior history with the moderator before this week and I do not wish to fight with them or continue any arguments here. I'm not interested in beef or drama. I'm just here for constructive discussion about Gmod development. 

I do not feel comfortable asking on the discord what would be an appropriate location to discuss CEF with the community as I believe I would be at risk of ban for mentioning CEF. I'd also like to ask if the developers are aware that the x86-64 branch did not receive the most recent update to the main branch and if that is intentional / bug-related / simply missed during the compile but again believe that I would risk a ban. So I am asking this here instead.
